### ISSUE TITLE
Remove deprecated API

### ISSUE THREAD

**Issue by arcturus2** [constellation-app/constellation#44]
### Prerequisites

* [x] Put an X between the brackets on this line if you have done all of the following:
    * Running the latest version of Constellation
	* Attached the ***Support Package*** via `Help > `Support Package`
    * Checked the FAQs: https://github.com/constellation-app/constellation/faq
    * Checked that your issue isn't already filed: https://github.com/constellation-app/constellation/issues
    * Checked that there is not already a module that provides the described functionality: https://github.com/constellation-app/constellation/repositories-catalog

### Description

* java datetime marked deprecated
* any deprecated Schema API
* any NetBeans API's that are deprecated

**Comment by cygnus-x-1** [comment]
The update to Java 11 seems to have brought with it a range of new deprecation warnings, in particular ```Object.newInstance``` and ```Object.finalize``` are now deprecated - I'll start looking into this now.

**Comment by arcturus2** [comment]
Great thank you @cygnus-x-1

**Comment by cygnus-x-1** [comment]
So far, I've manged to clean up all deprecation warnings except for two...

org.openide.util.actions.BooleanStateAction is deprecated and has been replaced with a new API which i'm having trouble figuring out.

java.lang.Object.finalize() is deprecated, but we are using it throughout the graph framework to remove references in MemoryManager - I'm not quite sure how to refactor that class to avoid using finalize just yet.

Does anyone have any thoughts?
